ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -600,49 +600,59 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 	}
 
 	// Reconcile head Pod
-	if len(headPods.Items) == 1 {
-		headPod := headPods.Items[0]
-		r.Log.Info("reconcilePods", "Found 1 head Pod", headPod.Name, "Pod status", headPod.Status.Phase,
-			"Pod restart policy", headPod.Spec.RestartPolicy,
-			"Ray container terminated status", getRayContainerStateTerminated(headPod))
-
-		shouldDelete, reason := shouldDeletePod(headPod, rayv1.HeadNode)
-		r.Log.Info("reconcilePods", "head Pod", headPod.Name, "shouldDelete", shouldDelete, "reason", reason)
-		if shouldDelete {
+
+	// if RayCluster is suspended, delete all head pods
```

Comment:
> It seems some logic is common to the standard cluster deletion.

I might have missed it, but I didn't see any deletion logic for the cluster, my assumption was that this was handled via OwnerReferences and Kubernetes garbage collection.

Reply:
It could have but I don't think that's the case. I'm referring to that code block in particular:

https://github.com/ray-project/kuberay/blob/4c2c046cc294bd5e24d675224f107890947d5a35/ray-operator/controllers/ray/raycluster_controller.go#L202-L259

Reply:
Right, sounds good!